src/CPlusPlusMangle.cpp

```diff
@@ -533,7 +533,11 @@ std::string mangle_type(const Type &type, const Target &target, PrevPrefixes &pr
         case 16:
             return "s";
         case 32:
-            return "i";
+            if (target.arch == Target::Hexagon) {
```

Comment:
Can you comment on what's going on here (and why we never noticed the issue before)?

Reply:
We never tested the name-mangling target feature for Hexagon and hence never saw the problem. I think the changes are needed because of Hexagon being a 32bit proc.

Reply:
If this is for 32-bit, shouldn't the condition be if (target.bits == 32) ?

But I think we test this code on other 32-bit targets, so I think this might be something else.

Reply:
@zvookin might know more

Reply:
@dsharletg  The headers have int32 defined as long and uint32 as unsigned long. That is why this change is needed on Hexagon.